### ISSUE TITLE
Mime Types for player.load()

### ISSUE THREAD

**Issue by hiren3897** [google/shaka-player#3561]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
yes 

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
yes

**What version of Shaka Player are you using?**
latest

**Please ask your question**
I have created a plugin that uses JSON URL and extracts the manifest URI from it and returns to play the content in the player.

But the problem is I have to manually pass the MimeType in `player.load` to tell the player about it.

I want to know what are the mime types for `hlsfairplay`,  Microsoft `play ready`, I know dash widevine have `application/dash+xml`  likewise what are the other mime-types that are accepted in shaka-player?

Is there any way to check the mime type of the current manifest loaded in the player?

Thank you in advance.

**Comment by TheModMaker** [comment]
FairPlay and PlayReady are DRM providers, they could be used with DASH too (although FairPlay is always HLS).  The MIME type for HLS is `application/x-mpegurl`.  You can also register the file extension so you don't have to pass it to `load`:

```js
shaka.media.ManifestParser.registerParserByExtension('json', () => new MyParser());
player.load('https://example.com/foo.json');
```

We also make a HEAD request to the URL and use the `Content-Type` header to pick the parser.  If the header uses the same MIME type you registered, we'll pick your parser.

I don't think there is a way to get the current MIME type, since usually apps just care about playing and the track list.  We abstract away the manifest type and convert it to a generic object format.

**Comment by hiren3897** [comment]
Hello @TheModMaker 
Thank you for your reply. 

For us, if we have a dash manifest or Microsoft play ready to manifest it has a `Content-Type: application/octet-stream` so can we pass this as a `MimeType` in the `player.load` 
or
by using 
```javascript
shaka.media.ManifestParser.registerParserByExtension('json', () => new MyParser());
player.load('https://example.com/foo.json');
```
I haven't tested for `hlsfairplay` but yes it accepts `application/x-mpegurl`.

My JSON returns a stream according to the client browser and OS device supported so if I want to register all 3 of them in the manifest parser how can I do that?

Example
if it returns dash then `application/dash+xml`, Microsoft then `application/dash+xml` or `application/octet-stream` and if apple fairplay then `application/x-mpegurl`

Thanks :)

**Comment by hiren3897** [comment]
@joeyparrish Yes, it helped me out a lot in my plugins. thank you for your great help!!
